### PR TITLE
feat: add v2 API for muteconfig Go client libraries or samples

### PR DESCRIPTION
**Pull request by vijaykanthm**
## Description

Reference to https://b.corp.google.com/issues/339692211
This PR adds v2 API to create, list, get, update, delete, mute, unmute, bulk-mute the samples. 

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist
- [ ] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [ ] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [ ] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: securitycenter/muteconfigv2/mute_config_test.go [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+		},
+	}
+	return client.CreateFinding(ctx, req)
+}
+
+func createSource(w io.Writer, orgId string) error {
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("securitycenter.NewClient: %v", err)
```
Comment (Deleplace) [inline securitycenter/muteconfigv2/mute_config_test.go:88]: Nit: I think `fmt.Errorf` is always better with the verb `%w` for an error, even in a test file  (this is different from `t.Errorf`)

4 occurrences

File: securitycenter/muteconfigv2/mute_config_test.go [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+func newMuteConfigFixture() (*muteconfigFixture, error) {
+	var mc muteconfigFixture
+	var err error
+
+	mc.client, err = securitycenter.NewClient(context.Background())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create security center client: %w", err)
+	}
+
+	orgId := os.Getenv("SCC_PROJECT_ORG_ID")
```
Comment (Deleplace) [inline securitycenter/muteconfigv2/mute_config_test.go:128]: Are these 2 env vars set in the test environment? Where do the values come from?

**Review by Deleplace — COMMENTED** [review]
Good code, well done.  PTAL at 2 minor comments

File: securitycenter/muteconfigv2/mute_config_test.go (hunk 1 above)
Comment (vijaykanthm) [reply securitycenter/muteconfigv2/mute_config_test.go]: Addressed.

File: securitycenter/muteconfigv2/mute_config_test.go (hunk 2 above)
Comment (vijaykanthm) [reply securitycenter/muteconfigv2/mute_config_test.go:128]: I will set it in the terminal while testing in local, but they will be defined in the CI to fetch the values and use them.

**Review by Deleplace — APPROVED** [review]
LGTM

**Pull request merged**